cmd/tracee-ebpf/capabilities.go

```diff
@@ -73,16 +75,53 @@ func getCapabilitiesRequiredByTraceeEvents(cfg *tracee.Config) []cap.Value {
 	return removeDupCaps(caps)
 }
 
+// Retrieve the value of the kernel parameter perf_event_paranoid
+func getKernelPerfEventParanoidValue() int {
+	/*
+	* perf event paranoia level:
+	*  -1 - not paranoid at all
+	*   0 - disallow raw tracepoint access for unpriv
+	*   1 - disallow cpu events for unpriv
+	*   2 - disallow kernel profiling for unpriv
+	*   4 - disallow all unpriv perf event use
+	 */
+	const MAX_PARANOIA_LEVEL = 4
+	value, err := os.ReadFile("/proc/sys/kernel/perf_event_paranoid")
+	if err != nil {
+		return MAX_PARANOIA_LEVEL
+	}
+	intVal, err := strconv.ParseInt(strings.TrimSuffix(string(value), "\n"), 0, 16)
+	if err != nil {
+		return MAX_PARANOIA_LEVEL
+	}
+	return int(intVal)
+}
+
 // Get all capabilities required for eBPF usage (including perf buffers maps management)
-func getCapabilitiesRequiredByEBPF(OSInfo KernelVersionInfo) ([]cap.Value, error) {
+func getCapabilitiesRequiredByEBPF(OSInfo KernelVersionInfo, debug bool) ([]cap.Value, error) {
 	// In kernel 5.8, CAP_BPF and CAP_PERFMON capabilities were introduced in order to replace CAP_SYS_ADMIN when
 	// loading eBPF programs.
-	// For some reasons, some distributions using new kernels still need CAP_SYS_ADMIN,
-	// so tracee still use it instead of the new capabilities.
-	caps := []cap.Value{
-		cap.IPC_LOCK,
-		cap.SYS_RESOURCE,
-		cap.SYS_ADMIN,
+	kernelParanoidValue := getKernelPerfEventParanoidValue()
+	var caps []cap.Value
+	// if kernelParanoidValue is too high, CAP_SYS_ADMIN is required
+	if kernelParanoidValue > 3 {
```

Comment:
You miss here the check of the kernel version, which should be at least 5.8 to use `CAP_BPF` and `CAP_PERFMON` (they don't exist before this version).

Should probably look the following:
```suggestion
        const bpfCapabilitiesMinKernelVersion = "5.8"
        if OSInfo.CompareOSBaseKernelRelease(bpfCapabilitiesMinKernelVersion) <= 0 {
	    if kernelParanoidValue > 3 {
```

Reply:
I will add that when I implemented it, I checked when using `CAP_BPF` and `CAP_PERFMON` if they are available.
If they aren't, I reverted to using `CAP_SYS_ADMIN`.
I think we try at all cost to make tracee work when used, so if there is a way to run it with even higher capabilities then we will use it.
You can see my implementation in commit 6b0cad48.

Reply:
Thank you, how can I modify the tests ?